### ISSUE TITLE
texture format utility

### ISSUE THREAD

**Issue by pollend** [NVIDIAGameWorks/NRI#22]
There is a selection of formats but sometimes you want to know how many samples there are or the bit depth of the format. would by nice to have some utilities similar to tiny_imageformat? 

https://github.com/DeanoC/tiny_imageformat/blob/master/include/tiny_imageformat/tinyimageformat.h

**Comment by dzhdanNV** [comment]
Hi! Thanks for your suggestion, it does make sense. There is `const FormatProps& GetFormatProps(nri::Format format)` ([link](https://github.com/NVIDIAGameWorks/NRI/blob/main/Source/Shared/SharedExternal.cpp#L478)) which is used internally and returns:
- stride between pixels in bytes
- block width in pixels
- "is integer" flag

It can be extended to include:
- NRI name as `const char*`
- "is normalized" flag
- "is float" flag
- "is signed" flag
- "is sRGB" flag
- "is compressed" flag
- channel count
This can be exposed. Which other data bits are you interested in? TIF is nice (at least in terms of functionality). If more is needed why not use TIF? Sample count requires querying... and that's the code, which I would prefer to not have.

**Comment by dzhdanNV** [comment]
Some device specific texture properties are exposed in `DeviceDesc`:
```
    // Multi-sampling
    NRI_NAME(Sample_t) colorSampleMaxNum;
    NRI_NAME(Sample_t) depthSampleMaxNum;
    NRI_NAME(Sample_t) stencilSampleMaxNum;
    NRI_NAME(Sample_t) zeroAttachmentsSampleMaxNum;
    NRI_NAME(Sample_t) textureColorSampleMaxNum;
    NRI_NAME(Sample_t) textureIntegerSampleMaxNum;
    NRI_NAME(Sample_t) textureDepthSampleMaxNum;
    NRI_NAME(Sample_t) textureStencilSampleMaxNum;
    NRI_NAME(Sample_t) storageTextureSampleMaxNum;
```

**Comment by pollend** [comment]
you have a good point I've kind of rolled out my own code. that is good to know, I'll have to think about it.

**Comment by dzhdanNV** [comment]
@pollend Hi! I have implemented (?) your request in `v1.133`, because @vertver also requested it, and I realized the usefulness. This info is available now:
```c++
NRI_STRUCT(FormatProps)
{
    const char* name;            // format name
    NRI_NAME(Format) format;     // self
    uint8_t redBits;             // R (or depth) bits
    uint8_t greenBits;           // G (or stencil) bits (0 if channels < 2)
    uint8_t blueBits;            // B bits (0 if channels < 3)
    uint8_t alphaBits;           // A (or shared exponent) bits (0 if channels < 4)
    uint32_t stride         : 6; // block size in bytes
    uint32_t blockWidth     : 4; // 1 for plain formats, >1 for compressed
    uint32_t blockHeight    : 4; // 1 for plain formats, >1 for compressed
    uint32_t isBgr          : 1; // reversed channels (RGBA => BGRA)
    uint32_t isCompressed   : 1; // block-compressed format
    uint32_t isDepth        : 1; // has depth component
    uint32_t isExpShared    : 1; // shared exponent in alpha channel
    uint32_t isFloat        : 1; // floating point
    uint32_t isPacked       : 1; // 16- or 32- bit packed
    uint32_t isInteger      : 1; // integer
    uint32_t isNorm         : 1; // [0; 1] normalized
    uint32_t isSigned       : 1; // signed
    uint32_t isSrgb         : 1; // sRGB
    uint32_t isStencil      : 1; // has stencil component
    uint32_t unused         : 7;
};
```
- I could have fit into 16 bytes (currently 24), but the current layout is compatible with 64-bits per channel formats and mystical ASTC compression
- let me know if something is missing
- please, validate my mega-table, I could make a mistake ;)
